core/amber/src/main/scala/edu/uci/ics/texera/web/TexeraWebApplication.scala

```diff
@@ -193,4 +215,53 @@ class TexeraWebApplication extends io.dropwizard.Application[TexeraWebConfigurat
     environment.jersey.register(classOf[DashboardResource])
   }
 
+  /**
+    * This function checks the executions that are not completed, meaning they were not closed gracefully because this function is called right after starting the server
+    * so no new workflows can be running
+    */
+  def cleanDirtyCollections(): Unit = {
+    // retrieve all executions that are not completed
+    val incompleteExecutions: List[WorkflowExecutionEntry] =
+      WorkflowExecutionsResource.getAllIncompleteResults()
+    // drop the collection and update the status to ABORTED
+    incompleteExecutions.foreach(execEntry => {
+      dropCollections(execEntry.result)
+      // then delete the pointer from mySQL
+      ExecutionsMetadataPersistService.tryUpdateExecutionStatusAndPointers(
+        execEntry.eId.longValue(),
+        WorkflowAggregatedState.ABORTED
+      )
+    })
+  }
+
+  def dropCollections(result: String): Unit = {
+    // parse the JSON
+    val node = objectMapper.readTree(result)
+    val collections = node.get("results")
+    // loop every collection and drop it
+    collections.forEach(collection => MongoDatabaseManager.dropCollection(collection.asText()))
+  }
+
+  /**
+    * This function is called periodically and checks all expired collections and deletes them
+    * MongoDB doesn't have an API of drop collection where collection name in (from a subquery), so the implementation is to retrieve
+    * the entire list of those documents that have expired, then loop the list to drop them one by one
+    */
+  def cleanOldCollections(
+      timeToLive: Int
+  ): Unit = {
+    // retrieve all executions that are completed and their last update time goes beyond the ttl
+    val expiredResults: List[ExecutionResultEntry] =
+      WorkflowExecutionsResource.getExpiredResults(timeToLive)
+    // drop the collection and update the status to ABORTED
+    expiredResults.foreach(execEntry => {
+      dropCollections(execEntry.result)
```

Comment:
Both `cleanDirtyCollections` and `cleanOldCollections` contain duplicated code for dropping the collection and set the status to ABORTED, we should refactor this logic into another function.

Reply:
Redundant code is removed.